test/test-tcp-connect-error-busy.c

```diff
@@ -0,0 +1,62 @@
+/* Copyright Joyent, Inc. and other Node contributors. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "uv.h"
+#include "task.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+
+static void connect_cb(uv_connect_t* handle, int status) {
+    FATAL("connect callback should not have been called!");
+}
+
+
+TEST_IMPL(tcp_connect_error_busy) {
+    struct sockaddr_in addr;
+    uv_tcp_t server;
+    int r;
+    uv_connect_t req;
+    uv_connect_t* connect_req = malloc(sizeof *connect_req);
```

Comment:
The purpose of initializing it, is to get the `NULL` check in [src/unix/tcp.c#L155](https://github.com/libuv/libuv/pull/684/files#diff-8d6b1362f259d05a583a404a2119a48dL155) to trigger:

```c
if (handle->connect_req != NULL)
  return -EBUSY;
```

I'm all ears to better ways of doing that.